lambda/matching.ml

```diff
@@ -1818,702 +2037,757 @@ let expand_stringswitch loc arg sw d = match d with
 let handle_shared () =
   let hs = ref (fun x -> x) in
-  let handle_shared act = match act with
-  | Switch.Single act -> act
-  | Switch.Shared act ->
-      let i,h = make_catch_delayed act in
-      let ohs = !hs in
-      hs := (fun act -> h (ohs act)) ;
-      make_exit i in
-  hs,handle_shared
-
+  let handle_shared act =
+    match act with
+    | Switch.Single act -> act
+    | Switch.Shared act ->
+        let i, h = make_catch_delayed act in
+        let ohs = !hs in
+        (hs := fun act -> h (ohs act));
+        make_exit i
+  in
+  (hs, handle_shared)
 
 let share_actions_tree sw d =
   let store = StoreExp.mk_store () in
-(* Default action is always shared *)
+  (* Default action is always shared *)
   let d =
     match d with
     | None -> None
-    | Some d -> Some (store.Switch.act_store_shared () d) in
-(* Store all other actions *)
+    | Some d -> Some (store.Switch.act_store_shared () d)
+  in
+  (* Store all other actions *)
   let sw =
-    List.map  (fun (cst,act) -> cst,store.Switch.act_store () act) sw in
-
-(* Retrieve all actions, including potential default *)
+    List.map (fun (cst, act) -> (cst, store.Switch.act_store () act)) sw
+  in
+  (* Retrieve all actions, including potential default *)
   let acts = store.Switch.act_get_shared () in
-
-(* Array of actual actions *)
-  let hs,handle_shared = handle_shared () in
+  (* Array of actual actions *)
+  let hs, handle_shared = handle_shared () in
   let acts = Array.map handle_shared acts in
-
-(* Reconstruct default and switch list *)
-  let d = match d with
-  | None -> None
-  | Some d -> Some (acts.(d)) in
-  let sw = List.map (fun (cst,j) -> cst,acts.(j)) sw in
-  !hs,sw,d
+  (* Reconstruct default and switch list *)
+  let d =
+    match d with
+    | None -> None
+    | Some d -> Some acts.(d)
+  in
+  let sw = List.map (fun (cst, j) -> (cst, acts.(j))) sw in
+  (!hs, sw, d)
 
 (* Note: dichotomic search requires sorted input with no duplicates *)
-let rec uniq_lambda_list sw = match sw with
-  | []|[_] -> sw
-  | (c1,_ as p1)::((c2,_)::sw2 as sw1) ->
-      if const_compare c1 c2 = 0 then uniq_lambda_list (p1::sw2)
-      else p1::uniq_lambda_list sw1
+let rec uniq_lambda_list sw =
+  match sw with
+  | []
+  | [ _ ] ->
+      sw
+  | ((c1, _) as p1) :: ((c2, _) :: sw2 as sw1) ->
+      if const_compare c1 c2 = 0 then
+        uniq_lambda_list (p1 :: sw2)
+      else
+        p1 :: uniq_lambda_list sw1
 
 let sort_lambda_list l =
-  let l =
-    List.stable_sort (fun (x,_) (y,_) -> const_compare x y) l in
+  let l = List.stable_sort (fun (x, _) (y, _) -> const_compare x y) l in
   uniq_lambda_list l
 
 let rec cut n l =
-  if n = 0 then [],l
-  else match l with
-    [] -> raise (Invalid_argument "cut")
-  | a::l -> let l1,l2 = cut (n-1) l in a::l1, l2
+  if n = 0 then
+    ([], l)
+  else
+    match l with
+    | [] -> raise (Invalid_argument "cut")
+    | a :: l ->
+        let l1, l2 = cut (n - 1) l in
+        (a :: l1, l2)
```

Comment:
Except for the `Invalid_argument` (and the behavior on negative integers), isn't this `cut` function the same as `rev_split_at`?

Reply:
Looks like it is indeed.